Sources/xcodeproj/Objects/Files/PBXFileElement.swift

```diff
@@ -141,15 +145,14 @@ public extension PBXFileElement {
     /// - Returns: file element absolute path.
     /// - Throws: an error if the absolute path cannot be obtained.
     public func fullPath(sourceRoot: Path) throws -> Path? {
-        let projectObjects = try objects()
         switch sourceTree {
         case .absolute?:
             return path.flatMap { Path($0) }
         case .sourceRoot?:
             return path.flatMap { sourceRoot + $0 }
         case .group?:
-            guard let group = projectObjects.groups.first(where: { $0.value.childrenReferences.contains(reference) }) else { return sourceRoot }
-            guard let groupPath = try group.value.fullPath(sourceRoot: sourceRoot) else { return nil }
+            guard let group = parent else { return sourceRoot }
+            guard let groupPath = try group.fullPath(sourceRoot: sourceRoot) else { return nil }
```

Comment:
I'd test the following scenarios:
- A file element that has a parent. Test that we get the right path.
- A file element that doesn't have a parent. Test that we get the right path.

Reply:
Ready